src/Crypto/Encoding/BIP39.hs

```diff
@@ -50,54 +49,63 @@ module Crypto.Encoding.BIP39
     , wordIndex
     , unWordIndex
 
+    -- * Errors
+    , BIP39Err(..)
+
     , -- * helpers
       ConsistentEntropy
     , CheckSumBits
     , Elem
     ) where
 
-import Prelude ((-), (*), (+), div, divMod, (^), fromIntegral)
+import           Prelude                          (div, divMod, fromIntegral,
+                                                   (*), (+), (-), (^))
 
-import           Basement.String (String)
-import qualified Basement.String as String
-import           Basement.Nat
-import qualified Basement.Sized.List as ListN
-import           Basement.NormalForm
 import           Basement.Compat.Typeable
-import           Basement.Numerical.Number (IsIntegral(..))
 import           Basement.Imports
+import           Basement.Nat
+import           Basement.NormalForm
+import           Basement.Numerical.Number        (IsIntegral (..))
+import qualified Basement.Sized.List              as ListN
+import           Basement.String                  (String)
+import qualified Basement.String                  as String
 
 import           Foundation.Check
 
-import           Control.Monad (replicateM)
+import           Control.Monad                    (replicateM)
 import           Data.Bits
-import           Data.Maybe (fromMaybe)
-import           Data.List (reverse)
-import           Data.ByteArray (ByteArrayAccess, ByteArray)
-import qualified Data.ByteArray as BA
-import           Data.ByteString (ByteString)
-import qualified Data.ByteString as BS
+import           Data.ByteArray                   (ByteArray, ByteArrayAccess)
+import qualified Data.ByteArray                   as BA
+import           Data.ByteString                  (ByteString)
+import qualified Data.ByteString                  as BS
+import           Data.Either
+import           Data.List                        (reverse)
 
 import           Data.Proxy
 
-import           GHC.Exts (IsList(..), IsString)
+import           GHC.Exts                         (IsList (..), IsString)
 
-import           Crypto.Hash (hashWith, SHA256(..))
-import           Crypto.Number.Serialize (os2ip, i2ospOf_)
-import qualified Crypto.KDF.PBKDF2 as PBKDF2
+import           Crypto.Hash                      (SHA256 (..), hashWith)
+import qualified Crypto.KDF.PBKDF2                as PBKDF2
+import           Crypto.Number.Serialize          (i2ospOf_, os2ip)
 
+import           Crypto.Encoding.BIP39.BIP39Err
 import           Crypto.Encoding.BIP39.Dictionary
```

Comment:
Not sure about all the import alignments. `cardano-crypto` doesn't use the same styleguide than `cardano-sl`.

Reply:
yes, we prefer to keep the history over some aesthetic changes.